skfp/distances/russell.py

```diff
@@ -0,0 +1,202 @@
+from typing import Union
+
+import numpy as np
+from scipy.sparse import csr_array
+from sklearn.utils._param_validation import validate_params
+
+
+@validate_params(
+    {
+        "vec_a": [
+            "array-like",
+            csr_array,
+        ],
+        "vec_b": [
+            "array-like",
+            csr_array,
+        ],
+    },
+    prefer_skip_nested_validation=True,
+)
+def russell_binary_similarity(
+    vec_a: Union[
+        np.ndarray,
+        csr_array,
+    ],
+    vec_b: Union[
+        np.ndarray,
+        csr_array,
+    ],
+) -> float:
+    r"""
+    Russell similarity for vectors of binary values.
+
+    Computes the Russell similarity [1]_ [2]_ [3]_ for binary data between two
+    input arrays or sparse matrices, using the formula:
+
+    .. math::
+
+        sim(x, y) = \frac{a}{n}
+
+    where
+
+    - :math:`a` - common "on" bits.
+    - :math:`n` - length of passed vectors
+
+    The calculated similarity falls within the range :math:`[0, 1]`.
+    Passing all-zero vectors to this function results in a similarity of 0.
+    Passing empty vectors results in similarity of 0.
+
+    Parameters
+    ----------
+    vec_a : {ndarray, sparse matrix}
+        First binary input array or sparse matrix.
+
+    vec_b : {ndarray, sparse matrix}
+        Second binary input array or sparse matrix.
+
+    Returns
+    -------
+    similarity : float
+        Russell similarity between ``vec_a`` and ``vec_b``.
+
+    References
+    ----------
+    .. [1] `Russell P.F., Rao T.R.
+        "On habitat and association of species of anopheline larvae in south-eastern Madras. (1940)"
```

Comment:
Without `. (1940)` in the name. Journal is: `[Journal of the Malaria Institute of India](https://www.cabidigitallibrary.org/action/doSearch?do=Journal+of+the+Malaria+Institute+of+India), 1940, June, Vol. 3, No. 1, 153-178 pp.`